integration/spark/shared/src/main/java/io/openlineage/spark/agent/filters/BigQueryIntermediateJobFilter.java

```diff
@@ -0,0 +1,61 @@
+/*
+/* Copyright 2018-2025 contributors to the OpenLineage project
+/* SPDX-License-Identifier: Apache-2.0
+*/
+
+package io.openlineage.spark.agent.filters;
+
+import static io.openlineage.spark.agent.filters.EventFilterUtils.getLogicalPlan;
+
+import io.openlineage.spark.api.OpenLineageContext;
+import java.util.Optional;
+import java.util.UUID;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.spark.scheduler.SparkListenerEvent;
+import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan;
+import org.apache.spark.sql.execution.datasources.InsertIntoHadoopFsRelationCommand;
+
+@Slf4j
+public class BigQueryIntermediateJobFilter implements EventFilter {
+
+  private final OpenLineageContext context;
+
+  public BigQueryIntermediateJobFilter(OpenLineageContext context) {
+    this.context = context;
+  }
+
+  @Override
+  public boolean isDisabled(SparkListenerEvent event) {
+    Optional<LogicalPlan> logicalPlan = getLogicalPlan(context);
+
+    if (!logicalPlan.isPresent()) {
+      return false;
+    }
+
+    if (!(logicalPlan.get() instanceof InsertIntoHadoopFsRelationCommand)) {
+      return false;
+    }
+
+    InsertIntoHadoopFsRelationCommand command =
+        (InsertIntoHadoopFsRelationCommand) logicalPlan.get();
```

Comment:
using isPresent()->get() is considered as an anti-pattern, is it possible to use map().orElseGet()?

Reply:
the whole logic operating on InsertIntoHadoopFsRelationCommand could be in separate method

Reply:
applied changes proposed